features/step_definitions/date_filters.rb

```diff
@@ -0,0 +1,58 @@
+And('The list has no filters') do
+  @original_amount_of_books = total_amount_of_books
+end
+
+When('I fill the {string} filter') do |date_filter|
+  filter = process_filter(date_filter)
+  wait_for_element_to_be_ready(:xpath, filter[:xpath])
+  check_element_text(:xpath, '', filter[:xpath], true)
+  click(:xpath, filter[:xpath])
+  enter_text(:xpath, filter[:date], filter[:xpath])
+end
+
+Then('The book list will be updated') do
+  @filtered_amount_of_books = total_amount_of_books
+  expect(@filtered_amount_of_books).not_to eq(@original_amount_of_books)
+end
+
+And("There won't be books with publication date previous to the filter date") do
+  expect(@filtered_amount_of_books).to eq(15)
+end
+
+When('I fill the from_date with the latest publication date possible') do
+  calendar_button_xpath = '//*[@id="search_picker"]/div/div/div/button'
```

Comment:
in this case I do not recommend using xpath, since it is prone to changes, if possible you can write the date directly in the field without opening the calendar

Reply:
I agree with this. Although you did a great job with this, it is not required to automate the calendar feature, just the manual date filter :grin: